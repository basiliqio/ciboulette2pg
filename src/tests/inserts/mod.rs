use super::*;

mod inserts_success;
