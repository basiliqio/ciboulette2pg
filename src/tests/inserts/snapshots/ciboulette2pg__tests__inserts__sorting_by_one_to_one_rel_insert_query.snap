---
source: src/tests/inserts/mod.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
WITH "cte_peoples_insert_0" AS (
  INSERT INTO
    "public"."peoples" (
      "age",
      "first-name",
      "gender",
      "last-name",
      "twitter"
    )
  VALUES
    ($1, $2, $3, $4, $5) RETURNING *
),
"cte_peoples_data_1" AS (
  SELECT
    "cte_peoples_insert_0"."id" :: TEXT AS "id",
    "cte_peoples_insert_0"."id" AS "main_id",
    NULL :: TEXT AS "related_id",
    NULL :: TEXT AS "related_type",
    $6 :: TEXT AS "type",
    JSON_BUILD_OBJECT(
      $7,
      "cte_peoples_insert_0"."age",
      $8,
      "cte_peoples_insert_0"."first-name",
      $9,
      "cte_peoples_insert_0"."gender",
      $10,
      "cte_peoples_insert_0"."last-name",
      $11,
      "cte_peoples_insert_0"."twitter"
    ) AS "data",
    "cte_peoples_insert_0"."favorite_color" AS "rel_favorite_color",
    "cte_peoples_insert_0"."first-name" AS "sort_first-name"
  FROM
    "cte_peoples_insert_0"
),
"cte_rel_favorite_color_data_2" AS (
  SELECT
    "public"."favorite_color"."id" :: TEXT AS "id",
    "public"."favorite_color"."id" AS "main_id",
    "cte_peoples_data_1"."id" :: TEXT AS "related_id",
    $12 :: TEXT AS "related_type",
    $13 :: TEXT AS "type",
    NULL :: json AS "data",
    "public"."favorite_color"."color" AS "sort_color"
  FROM
    "public"."favorite_color"
    INNER JOIN "cte_peoples_data_1" ON "cte_peoples_data_1"."rel_favorite_color" = "public"."favorite_color"."id"
),
"cte_final_main_data" AS (
  SELECT
    DISTINCT ON ("cte_peoples_data_1"."main_id") "cte_peoples_data_1".*,
    "cte_peoples_data_1"."sort_first-name" AS "sort__first-name",
    "cte_rel_favorite_color_data_2"."sort_color" AS "sort_favorite_color_color"
  FROM
    "cte_peoples_data_1"
    LEFT JOIN "cte_rel_favorite_color_data_2" ON "cte_rel_favorite_color_data_2"."main_id" = "cte_peoples_data_1"."rel_favorite_color"
  ORDER BY
    "cte_peoples_data_1"."main_id"
) (
  SELECT
    "cte_final_main_data"."id",
    "cte_final_main_data"."type",
    "cte_final_main_data"."data",
    "cte_final_main_data"."related_type",
    "cte_final_main_data"."related_id"
  FROM
    "cte_final_main_data"
  ORDER BY
    "cte_final_main_data"."sort_favorite_color_color" DESC,
    "cte_final_main_data"."sort__first-name" ASC
);
